Documentation/network/kubernetes/kubeproxy-free.rst

```diff
@@ -1399,6 +1399,101 @@ As per `k8s Service <https://kubernetes.io/docs/concepts/services-networking/ser
 Cilium's eBPF kube-proxy replacement by default disallows access to a ClusterIP service from outside the cluster.
 This can be allowed by setting ``bpf.lbExternalClusterIP=true``.
 
+Observability
+*************
+
+You can trace socket LB related datapath events using Hubble and cilium monitor.
+
+Apply the following pod and service:
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: nginx
+      labels:
+        app: proxy
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:stable
+        ports:
+          - containerPort: 80
+    ---
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: nginx-service
+    spec:
+      selector:
+        app: proxy
+      ports:
+      - port: 80
+
+Deploy a client pod to start traffic.
+
+.. code-block:: shell-session
+
+    $ kubectl create -f \ |SCM_WEB|\/examples/kubernetes-dns/dns-sw-app.yaml
+
+.. code-block:: shell-session
+
+    $ kubectl get svc | grep nginx
+      nginx-service   ClusterIP   10.96.128.44   <none>        80/TCP    140m
+
+    $ kubectl exec -it mediabot -- curl -v --connect-timeout 5 10.96.128.44
+
+Follow the Hubble :ref:`hubble_cli` guide  to see the network flows. The Hubble
+output prints datapath events before and after socket LB translation between service
+and selected service endpoint.
+
+.. code-block:: shell-session
+
+    $ hubble observe --all | grep mediabot
+    Jan 13 13:47:20.932: default/mediabot (ID:5618) <> default/nginx-service:80 (world) pre-xlate-fwd TRACED (TCP)
+    Jan 13 13:47:20.932: default/mediabot (ID:5618) <> default/nginx:80 (ID:35772) post-xlate-fwd TRANSLATED (TCP)
+    Jan 13 13:47:20.932: default/nginx:80 (ID:35772) <> default/mediabot (ID:5618) pre-xlate-rev TRACED (TCP)
+    Jan 13 13:47:20.932: default/nginx-service:80 (world) <> default/mediabot (ID:5618) post-xlate-rev TRANSLATED (TCP)
+    Jan 13 13:47:20.932: default/mediabot:38750 (ID:5618) <> default/nginx (ID:35772) pre-xlate-rev TRACED (TCP)
+
+Socket LB tracing with Hubble requires cilium agent to detect pod cgroup paths.
+If you see a warning message in cilium agent "No valid cgroup base path found: socket load-balancing tracing with Hubble will not work.",
+you can trace packets using ``cilium monitor`` instead.
+
+.. note::
+
+    File a GitHub issue with an example cgroup path for any pod by running command
+    on a Kubernetes node in your cluster: ``sudo crictl inspectp -o=json $POD_ID | grep cgroupsPath``.
```

Comment:
You mean that the user should file the issue if their cgroup path fails to be detected, right?

```suggestion
    If the cgroup path is not found, please file a GitHub issue with the cgroup path for any of your pods, obtained by running the following command
    on a Kubernetes node in your cluster: ``sudo crictl inspectp -o=json $POD_ID | grep cgroupsPath``.
```